provers/cairo/src/runner/run.rs

```diff
@@ -122,22 +125,46 @@ pub fn run_program(
     let cairo_mem = CairoMemory::from_bytes_le(&memory_vec).unwrap();
     let register_states = RegisterStates::from_bytes_le(&trace_vec).unwrap();
 
-    let data_len = runner.get_program().data_len();
+    let codelen = runner.get_program().data_len();
+
+    let public_memory = (1..=codelen as u64)
+        .map(|i| (Felt252::from(i), *cairo_mem.get(&i).unwrap()))
+        .collect::<HashMap<Felt252, Felt252>>();
```

Comment:
Now that we have the public inputs directly from the cairo vm, there is no need to ask for the codelen, and the public memory should be obtained from the public inputs, this was just a hacky solution until we had access to them.
The approach should be similar to the one you did with the memory segments

Reply:
As a result of this, the `PublicInputs` field `codelen` could be removed, since it can be deduced with the length of the `Program` memory segment

Reply:
The only function that makes use of the `codelen` field is `get_memory_holes`, should we remove it?

Reply:
> The approach should be similar to the one you did with the memory segments

I'm having trouble doing an iteration over the vm public memory, doing something like this:
```rs
    let mut public_memory: HashMap<Felt252, Felt252> = HashMap::new();
    vm_public_inputs.public_memory.iter().for_each(|e| {
        let v = e.value.clone().unwrap().to_str_radix(16);
        public_memory.insert(
            Felt252::from((e.address + e.page) as u64),
            Felt252::from_hex_unchecked(&v),
        );
    });
   ```
   
Breaks some of our tests
   
![image](https://github.com/lambdaclass/lambdaworks/assets/62400508/72b2aa72-3345-49bb-86ec-b73c798cee7d)

Maybe I'm missing something?